Mapbox/MapboxMapsLocation/Pucks/LocationPuckManager.swift

```diff
@@ -25,10 +25,19 @@ public enum LocationPuck: Equatable {
 
     public static func == (lhs: LocationPuck, rhs: LocationPuck) -> Bool {
         switch (lhs, rhs) {
-        case (.puck2D(_), .puck2D(_)):
-            return true
-        case (.puck3D(_), .puck3D(_)):
-            return true
+        case (.puck2D(let lhsClosure), .puck2D(let rhsClosure)):
+            guard let lhsClosure = lhsClosure, let rhsClosure = rhsClosure else { return false }
+            var lhsViewModel = LocationIndicatorLayerViewModel()
+            var rhsViewModel = LocationIndicatorLayerViewModel()
+            lhsClosure(&lhsViewModel)
+            rhsClosure(&rhsViewModel)
+            return lhsViewModel == rhsViewModel
+        case (.puck3D(let lhsClosure), .puck3D(let rhsClosure)):
+            var lhsViewModel = PuckModelLayerViewModel()
+            var rhsViewModel = PuckModelLayerViewModel()
+            lhsClosure(&lhsViewModel)
+            rhsClosure(&rhsViewModel)
+            return lhsViewModel == rhsViewModel
```

Comment:
Invoking closures in `==` would be highly unexpected, and could have significant performance & correctness implications for client apps depending on whether the closures have any side-effects.

Some alternatives to consider:
* Define `==` in terms of the closure identity (`===`)
* Remove `LocationPuck`'s `Equatable` conformance

Reply:
Ah good point, I hadn't considered that calling the closure could cause an issue for clients. Looking into `===`.